adafruit_mprls.py

```diff
@@ -32,21 +32,105 @@
 
 **Hardware:**
 
-.. todo:: Add links to any specific hardware product page(s), or category page(s). Use unordered list & hyperlink rST
-   inline format: "* `Link Text <url>`_"
-
 **Software and Dependencies:**
 
 * Adafruit CircuitPython firmware for the supported boards:
   https://github.com/adafruit/circuitpython/releases
-  
-.. todo:: Uncomment or remove the Bus Device and/or the Register library dependencies based on the library's use of either.
 
-# * Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
-# * Adafruit's Register library: https://github.com/adafruit/Adafruit_CircuitPython_Register
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+
 """
 
 # imports
 
 __version__ = "0.0.0-auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_MPRLS.git"
+
+
+import time
+from adafruit_bus_device.i2c_device import I2CDevice
+from digitalio import Direction
+from micropython import const
+
+# pylint: disable=bad-whitespace
+_MPRLS_DEFAULT_ADDR = const(0x18)
+# pylint: enable=bad-whitespace
+
+class MPRLS:
+    """
+    Driver base for the MPRLS pressure sensor
+    :param i2c_bus: The `busio.I2C` object to use. This is the only required parameter.
+    :param int addr: The optional I2C address, defaults to 0x18
+    :param microcontroller.Pin reset_pin: Optional digitalio pin for hardware resetting
+    :param microcontroller.Pin eoc_pin: Optional digitalio pin for getting End Of Conversion signal
+    :param float psi_min: The minimum pressure in PSI, defaults to 0
+    :param float psi_max: The maximum pressure in PSI, defaults to 25
+    """
+
+    def __init__(self, i2c_bus, *, addr=_MPRLS_DEFAULT_ADDR,
+                 reset_pin=None, eoc_pin=None, psi_min=0, psi_max=25):
+        # Init I2C
+        self._i2c = I2CDevice(i2c_bus, addr)
+        self._buffer = bytearray(4)
+
+        # Optional hardware reset pin
+        if reset_pin is not None:
+            reset_pin.direction = Direction.OUTPUT
+            reset_pin.value = True
+            reset_pin.value = False
+            time.sleep(0.01)
+            reset_pin.value = True
+        time.sleep(0.005) # Start up timing
+
+        # Optional end-of-conversion pin
+        self._eoc = eoc_pin
+        if eoc_pin is not None:
+            self._eoc.direction = Direction.INPUT
+
+        if psi_min >= psi_max:
+            raise RuntimeError("Min PSI must be < max!")
```

Comment:
Nit-noid alert: shouldn't this be a ValueError?

Reply:
good idea, fixed!